libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -94,7 +97,12 @@ public static String getUniqueId(String name, int length) {
      * @return Encryption key.
      */
     public static String getEncryptionKey(String name) {
-        return generateEncryptionKey(name);
+        String encryptionKey = CACHED_ENCRYPTION_KEYS.get(name);
+        if (encryptionKey == null) {
+            encryptionKey = generateEncryptionKey(name);
```

Comment:
NB: despite the name it's not actually generating a new encryption key every time.
It checks if we have a stored unique id and derive the key from it.